core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -430,17 +432,13 @@ public void onToolClick(PlayerInteractEvent event) {
   Component getMessage(
       String messageKey, Map.Entry<UUID, ? extends Number> mapEntry, TextColor color) {
     return TranslatableComponent.of(
-        messageKey,
-        playerName(mapEntry.getKey()),
-        numberComponent(String.valueOf(mapEntry.getValue()), color));
-  }
-
-  public static Component numberComponent(int stat, TextColor color) {
-    return TextComponent.of(stat, color, TextDecoration.BOLD);
+        messageKey, playerName(mapEntry.getKey()), numberComponent(mapEntry.getValue(), color));
   }
 
-  public static Component numberComponent(String stat, TextColor color) {
-    return TextComponent.of(stat, color, TextDecoration.BOLD);
+  public static Component numberComponent(Number stat, TextColor color) {
+    double doubleStat = stat.doubleValue();
+    if (doubleStat % 1 == 0) return TextComponent.of((int) stat, color, TextDecoration.BOLD);
+    else return TextComponent.of((double) stat, color, TextDecoration.BOLD);
```

Comment:
Won't this create `3.333333333` things unless passed thru a number format?

Reply:
Its already passed through a format when its fetched from the `PlayerStats` 🤔  I'll check

Reply:
<img width="611" alt="Skjermbilde 2020-09-17 kl  18 49 54" src="https://user-images.githubusercontent.com/19822231/93502122-a02b2400-f916-11ea-970b-d8c10b4604d8.png">

<img width="516" alt="Skjermbilde 2020-09-17 kl  18 50 42" src="https://user-images.githubusercontent.com/19822231/93502182-b0db9a00-f916-11ea-9f1b-ed7a8759a728.png">

<img width="512" alt="Skjermbilde 2020-09-17 kl  18 50 37" src="https://user-images.githubusercontent.com/19822231/93502227-bc2ec580-f916-11ea-815c-435e4b1be771.png">

Had to do some small finicking to parse correctly, but should work as intended now 👍 
(Also fixed the fact damage to yourself is tracked in the last commit)